plugins/API/tests/Unit/WidgetMetadataTest.php

```diff
@@ -48,6 +48,7 @@ public function test_buildWidgetMetadata_ShouldGenerateMetadata()
                 'order' => 99,
                 'icon' => '',
                 'help' => '',
+                'widget' => null,
```

Comment:
Probably a good idea to test this with an actual value at least once.

As you are missing a test for this logic:
`'widget' => $category->getWidget() ?: null,`

Reply:
Yep, I can probably update one of the system tests where it returns the API response in XML format for comparison.

Otherwise the UI tests for the menu already cover the visual component override depending on this working.